qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -391,249 +317,80 @@ def error_message(self):
 
         return self._api_error_msg
 
-    def queue_position(self):
+    def queue_position(self) -> Optional[int]:
         """Return the position in the server queue.
 
         Returns:
-            Number: Position in the queue.
+            int: Position in the queue or ``None`` if position is unknown.
         """
+        # Get latest position
+        self.status()
         return self._queue_position
 
-    def creation_date(self):
-        """Return creation date."""
+    def creation_date(self) -> str:
+        """Return creation date.
+
+        Returns:
+            str: Job creation date.
+        """
         return self._creation_date
 
-    def job_id(self, timeout=60):
+    def job_id(self) -> str:
         """Return the job ID assigned by the API.
 
-        If the job ID is not set because the job is still initializing, this
-        call will block until a job ID is available or the timeout is reached.
-
-        Args:
-            timeout (float): number of seconds to wait for the job ID.
-
         Returns:
             str: the job ID.
         """
-        self._wait_for_submission(timeout)
         return self._job_id
 
-    def submit(self, job_name=None):
+    def submit(self) -> None:
         """Submit job to IBM-Q.
 
-        Args:
-            job_name (str): custom name to be assigned to the job.
+        Note:
+            This function waits for a job ID to become available.
 
         Events:
             ibmq.job.start: The job has started.
 
         Raises:
-            JobError: If we have already submitted the job.
+            JobError: If an error occurred during job submit.
         """
-        # TODO: Validation against the schema should be done here and not
-        # during initialization. Once done, we should document that the method
-        # can raise QobjValidationError.
-        if self._future is not None or self._job_id is not None:
+        if self.job_id() is not None:
```

Comment:
Can you check if we can remove this `if` (and a similar one at `.error_message()`)? Hopefully we can now guarantee that all jobs have an `id`.

Reply:
This is to catch the off chance someone creates an `IBMQJob` manually and try to submit it.